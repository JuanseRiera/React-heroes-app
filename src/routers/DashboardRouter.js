import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { MarvelScreen } from "../components/Marvel/MarvelScreen";
import { HeroScreen } from "../components/heroes/HeroScreen";
import { DCScreen } from "../components/DC/DCScreen";
import { Navbar } from "../components/ui/NavBar";
import { SearchScreen } from "../components/search/SearchScreen";
export const DashboardRoutes = () => {
  return (
    <>
      <Navbar />

      <div className="container mt-5">
        <Switch>
          <Route exact path="/marvel" component={MarvelScreen} />
          <Route exact path="/dc" component={DCScreen} />
          <Route exact path="/search" component={SearchScreen} />
          <Route exact path="/hero/:heroeId" component={HeroScreen} />

          <Redirect to="/marvel" />
        </Switch>
      </div>
    </>
  );
};
