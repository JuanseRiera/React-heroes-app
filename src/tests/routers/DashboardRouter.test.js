import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRouter";

describe("Pruebas del componete DashboardRouter", () => {
  test("Deberia de cargar el componete", () => {
    const authValue = {
      user: {
        logged: true,
        name: "Juanse",
      },
      dispatch: jest.fn(),
    };
    const wrapper = mount(
      <AuthContext.Provider value={authValue}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("span").text()).toBe("Juanse");
  });
});
