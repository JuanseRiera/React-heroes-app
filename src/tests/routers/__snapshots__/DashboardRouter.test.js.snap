// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componete DashboardRouter Deberia de cargar el componete 1`] = `ReactWrapper {}`;
